enhancements/installer/azure-internal-with-user-defined-routing.md

```diff
@@ -0,0 +1,243 @@
+2. Ensure `UserDefinedRouting` is only used when creating a cluster with `Internal` publishing strategy. 
+
+### Kube cloud config
+
+When the `outboundType` is set to `UserDefinedRouting`, make sure that the `disableOutboundSNAT` field for the kube cloud config should be set to false.
+
+Setting `disableOutboundSNAT` to false ensures that any new public standard loadbalancer that is created by Kubernetes service type Loadbalancer does not enable egress through it, creating weird routing topologies.
+
+### Loadbalancer
+
+#### when is public IPv4 address required?
+
+- it's required for External clusters
+- it's required for Internal clusters when outboundType is Loadbalancer
+- it's required for IPv6 clusters because AzureRM LoadBalancers cannot have only IPv6 frontend :/
+
+#### when is public IPv6 address required?
+
+- it's required for External IPv6 clusters
+- it's required for Internal clusters when outboundType is Loadbalancer
+
+#### azure_lb_rules
+
+- the k8s API rules are required for External clusters
+- the k8s API rules provide SNAT only when outboundType is Loadbalancer
+- the k8s API rules are not required for Internal clusters
+- the outbound rules (dummy rules) are required for Internal clusters outboundType is Loadbalancer
+
+#### azure_lb_backends
+
+The backends are only created when the frontend configurations are created because of the failure from Azure
+```
+Load Balancer /subscriptions/xx/resourceGroups/xx/providers/Microsoft.Network/loadBalancers/xx-public-lb does not have Frontend IP Configuration, but it has other child resources. This setup is not supported.
+```
+
+Since the backends are not created for certain cases, the master and bootstrap modules need to skip adding the virtual machines to the azure_lb backends. Although it should be simple to switch on whether the backend was created i.e null/vs not null, the terraform [issue](https://github.com/hashicorp/terraform/issues/12570) doesn't allow such conditional in count.
+
+```
+ERROR Error: Invalid count argument
+ERROR
+ERROR   on ../../../../../../../tmp/openshift-install-941276375/bootstrap/main.tf line 142, in resource "azurerm_netwo
+ERROR  142:   count = var.elb_backend_pool_v4_id == null ? 0 : 1
+ERROR
+ERROR The "count" value depends on resource attributes that cannot be determined
+ERROR until apply, so Terraform cannot predict how many instances will be created.
+ERROR To work around this, use the -target argument to first apply only the
+ERROR resources that the count depends on.
+```
+
+And therefore, the master and bootstrap modules need to recreate the conditions of `need_public_ipv{4,6}` using the inputs `use_ipv{4,6}`, `private`, `outbound_udr`.
+
+#### azure_lb
+
+The public loadbalancer is always created to keep the complexity to the minimum. The public IPs, frontends, rules and backends will not be attached when necessary. A loadbalancer with no rules is already free in Azure so it should not add any extra cost to the user.
```

Comment:
I expect there'll be some push back on this from ARO customers/MSFT SAs.  People would probably like to be able to say "oh, there's no public LB, therefore it's provably private".